### PR TITLE
Bump to 2.3.1 - https://github.com/spotify/luigi/releases/tag/2.3.1

### PR DESCRIPTION
**Pull request by kwilcox**

### CODE REVIEW

**Comment by ocefpaf** [comment]
@kwilcox can you rerender the recipe to see if the failure goes away. I don't have that in my local build system because I dropped `conda-build-setup`. The `conda-build-setup` adds a extra layer to manage the conda build tools that masks what is happening there.

**Comment by kwilcox** [comment]
```
conda-build               2.0.0                    py34_0    defaults
conda-smithy              1.1.2                    py34_1    conda-forge
```

```
$ conda-smithy rerender          
No circle token.  Create a token at https://circleci.com/account/api and
put it in ~/.conda-smithy/circle.token
No appveyor token. Create a token at https://ci.appveyor.com/api-token and
Put one in ~/.conda-smithy/appveyor.token
No anaconda token. Create a token via
  anaconda auth --create --name conda-smithy --scopes "repos conda api"
and put it in ~/.conda-smithy/anaconda.token
Traceback (most recent call last):
  File "/home/kwilcox/miniconda3/bin/conda-smithy", line 9, in <module>
    load_entry_point('conda-smithy==1.1.2', 'console_scripts', 'conda-smithy')()
  File "/home/kwilcox/miniconda3/lib/python3.4/site-packages/conda_smithy/cli.py", line 236, in main
    args.subcommand_func(args)
  File "/home/kwilcox/miniconda3/lib/python3.4/site-packages/conda_smithy/cli.py", line 185, in __call__
    configure_feedstock.main(args.feedstock_directory)
  File "/home/kwilcox/miniconda3/lib/python3.4/site-packages/conda_smithy/configure_feedstock.py", line 399, in main
    conda_build.metadata.config.CONDA_NPY = '99.9'
AttributeError: 'module' object has no attribute 'config'
```

**Comment by kwilcox** [comment]
Seems to work with `conda-build<2.0` locally

File: .travis.yml [hunk 1]
```diff
@@ -28,12 +30,10 @@ install:
       bash $MINICONDA_FILE -b
 
       export PATH=/Users/travis/miniconda3/bin:$PATH
-
-      conda config --set show_channel_urls true
-      conda update --yes conda
-      conda install --yes conda-build=1.20.0 jinja2 anaconda-client
```
Comment (ocefpaf) [inline .travis.yml]: That is why OS C was passing.

File: .travis.yml [hunk 2]
```diff
@@ -28,12 +30,10 @@ install:
...
       conda config --add channels conda-forge
-      
+      conda config --set show_channel_urls true
+      conda install --yes --quiet conda-forge-build-setup
```
Comment (ocefpaf) [inline .travis.yml]: Those versions are now defined here. (And hidden from the user :unamused:)

File: ci_support/run_docker_build.sh [hunk 3]
```diff
@@ -39,9 +38,8 @@ echo "$config" > ~/.condarc
 # A lock sometimes occurs with incomplete builds. The lock file is stored in build_artefacts.
 conda clean --lock
 
-conda update --yes --all
-conda install --yes conda-build
-conda info
+conda install --yes --quiet conda-forge-build-setup
```
Comment (ocefpaf) [inline ci_support/run_docker_build.sh]: So on Linux things may pass as long as `conda-forge-build-setup` is properly pinned.

(That is why I no longer use `conda-forge-build-setup` locally. It hides the build tool versions and makes it harder for the end user to know what is going on.)

**Comment by ocefpaf** [comment]
It looks like `luigi` needs `setuptools` as a run-time dependency.

```shell
+ /bin/bash -x -e /feedstock_root/build_artefacts/test-tmp_dir/run_test.sh
+ luigid --help
Traceback (most recent call last):
  File "/opt/conda/envs/_test/bin/luigid", line 6, in <module>
    sys.exit(luigi.cmdline.luigid())
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/luigi/cmdline.py", line 15, in luigid
    import luigi.server
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/luigi/server.py", line 49, in <module>
    import pkg_resources
ImportError: No module named pkg_resources
TESTS FAILED: luigi-2.3.1-py27_0
```


(That is another decision I don't have to do locally as I decided to ship Python with `setuptools` by default as it is done with python.org binaries.)

**Comment by ocefpaf** [comment]
@kwilcox is Windows still a `luigi` limitation?

**Comment by kwilcox** [comment]
I'm not entirely sure. I don't see why it would be if there are windows packges for `tornado` and `python-daemon` available.

**Comment by ocefpaf** [comment]
OK `python-daemon` is the issue there then.

**Comment by kwilcox** [comment]
I imagine only a fraction of people even use the `--background` option for `luigid` and its only imported when the actual call is made: https://github.com/spotify/luigi/blob/7db1ceadc4d3f8ef62eccffa1a6412bc747f8132/luigi/process.py#L79

Maybe we can just make a note of this and build for windows as well?

**Comment by ocefpaf** [comment]
> its only imported when the actual call is made

That is a green light for optional dependency IMO.

> Maybe we can just make a note of this and build for windows as well?

Agreed. Let's just add the `# [not win]` for the `python-daemon` and build it on Windows.

**Comment by ocefpaf** [comment]
I can add the Windows build in another PR to keep the scope of this one.

**Pull request merged**